tests/NuGet.IndexingTests/TestSupport/MockSearcherManager.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using NuGet.Indexing;
+
+namespace NuGet.IndexingTests.TestSupport
+{
+    public class MockSearcherManager : NuGetSearcherManager
+    {
+        private string _indexName;
+
+        public MockSearcherManager(string indexName) 
+            : base(indexName: indexName, 
+                  logger: MockObjectFactory.CreateMockLogger().Object, 
+                  directory: null, 
+                  loader: null, 
+                  azureDirectorySynchronizer: null)
+        {
+            _indexName = indexName;
+            RegistrationBaseAddress[Constants.SchemeName] = new Uri(Constants.BaseUri);
```

Comment:
I want the constructor for this object to hide the fact that we need more stuff since we don't care about it in the test. This constructor also sets the RegistrationBaseAddress at construction time, which I also don't want the tests to explicitly set (just know what it is supposed to be given a scheme). I think @scottbommarito commented on this as well. I think there should be a discussion above.